src/map.h

```diff
@@ -247,6 +247,16 @@ class Map
 		bool canThrowObjectTo(const Position& fromPos, const Position& toPos, bool checkLineOfSight = true,
 		                      int32_t rangex = Map::maxClientViewportX, int32_t rangey = Map::maxClientViewportY) const;
 
+		/**
+		  * Checks if there are no obstacles on that position
+		  *	\param blockFloor counts the ground tile as an obstacle
+		  *	\returns The result if there is an obstacle or not
+		  */
+		bool isTileClear(uint16_t x, uint16_t y, uint8_t z, bool blockFloor = false) const;
+
+		bool getSteepLine(uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1, uint8_t z) const;
+		bool getSlightLine(uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1, uint8_t z) const;
+
```

Comment:
I don't think these belong here. They can be in an anonymous namespace only in the .cpp file for Very Aggressive Optimization :tm:

Reply:
I can't access `isTileClear`. It asks me for `Map` object.

Reply:
There should be no problem, it's direct call to another class